### PR TITLE
Adding a sample project

### PR DESCRIPTION
**Pull request by assyadh**
*Issue #, if available:* N/A

*Description of changes:*

This sample project is based on the VS2017 template for a 2.1 ASP.NET Core webapp.

The only files needed for review are:

ConfirmAccount.cshtml

Login.cshtml.cs

LoginWith2fa.cshtml.cs

Logout.cshtml.cs

Register.cshtml.cs

They include really small changes to use Cognito, such as referencing the CognitoUser of using the specific managers to access some Cognito specific signatures.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        public async Task<IActionResult> OnPostAsync(string returnUrl = null)
+        {
+            returnUrl = returnUrl ?? Url.Content("~/");
+            if (ModelState.IsValid)
+            {
+                var userId = User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Name).Value;
+
+                var user = await _userManager.FindByIdAsync(userId).ConfigureAwait(false);
```
Comment (costleya) [inline samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs]: We don't need ConfigureAwait. This is a webapp, and a netcore one at that,

File: samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            _userManager = userManager as CognitoUserManager<CognitoUser>;
+        }
+
+        [BindProperty]
+        public InputModel Input { get; set; }
+
+        public string ReturnUrl { get; set; }
+
+        public class InputModel
```
Comment (costleya) [inline samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs]: This is a small model, yes, but it is an independent part of MVC. It should be given it's own file.

File: samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                }
+
+                var result = await _userManager.ConfirmSignUpAsync(user, Input.Code, true).ConfigureAwait(false);
+                if (!result.Succeeded)
+                {
+                    throw new InvalidOperationException($"Error confirming account for user with ID '{userId}':");
+                }
+                else
+                {
+                    if (returnUrl != null)
```
Comment (costleya) [inline samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs]: return returnUrl != null ? LocalRedirect(returnUrl) : Page();

File: samples/Samples/Areas/Identity/Pages/Account/Login.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+            if (ModelState.IsValid)
+            {
+                var result = await _signInManager.PasswordSignInAsync(Input.UserName, Input.Password, Input.RememberMe, lockoutOnFailure: false).ConfigureAwait(false);
+                if (result.Succeeded)
+                {
+                    _logger.LogInformation("User logged in.");
+                    return LocalRedirect(returnUrl);
+                }
+                if (result.RequiresTwoFactor)
```
Comment (costleya) [inline samples/Samples/Areas/Identity/Pages/Account/Login.cshtml.cs]: I know all of these if statements have returns, but adding "else if" instead would improve maintainability.

File: samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs (hunk 2 above)
Comment (assyadh) [reply samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs]: https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-2.2&tabs=visual-studio

File: samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs (hunk 3 above)
Comment (assyadh) [reply samples/Samples/Areas/Identity/Pages/Account/ConfirmAccount.cshtml.cs]: LocalRedirect() and Page() do not return the same type

**Pull request merged**